src/client/tile.cpp

```diff
@@ -1012,11 +1012,11 @@ video::IImage * Align2Npot2(video::IImage * image,
 
 	core::dimension2d<u32> dim = image->getDimension();
 
-	std::string extensions = (char*) glGetString(GL_EXTENSIONS);
+	const char *extensions = glGetString(GL_EXTENSIONS);
 
 	// Only GLES2 is trusted to correctly report npot support
-	if (get_GL_major_version() > 1 &&
-			extensions.find("GL_OES_texture_npot") != std::string::npos) {
+	if (get_GL_major_version() > 1 && extensions != NULL &&
+			strstr(extensions, "GL_OES_texture_npot") != NULL) {
```

Comment:
` != NULL` is superfluous, pointer-to-bool conversion is well-defined. If you want such syntax anyway, use `nullptr` at least.

Reply:
will nullptr work with a C function call which doesn't know nullptr_t ?

Reply:
That doesn’t matter. Null pointer is a null pointer, regardless of how you name it. The reason to use `nullptr` is that it is only compatible with pointers, while `NULL` may (and is, in glibc at least) be defined as literal `0`.

Actually, the only allowed definitions for `NULL` are `nullptr` and `0` in C++. `((void*)0)` may only work in C as C++ forbids most implicit pointer conversions.

P.S. Sorry for OT.

Reply:
No problem, is the new version okay to you ?